tests/bluetooth/ctrl_user_ext/src/ull_vendor.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2019 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_BT_CTLR_USER_EXT)
+
+/* The test is made to show that ull.c and other code can compile when
+ * certain Kconfig options are set. This includes functions that are
+ * used for vendor specific behavior, as such the function implementations
+ * in this file are simple stubs without functional behavior.
+ */
+
+static inline int ull_user_init(void)
+{
+	return 0;
+}
+
+static inline int rx_demux_rx_proprietary(memq_link_t *link,
```

Comment:
Any idea how the user will get access to the memq.h file when implementing this? Or does it always have to be implemented as static inline in the header file?

Reply:
Static inline is not required, but I wanted to show that static inline can indeed be used as this would result in almost the same compiled code as if the functional behavior was written in the canonical files

Reply:
I think you misunderstood my point. The prototype for rx_demux_rx_proprietary depends on memq_link_t which is defined in memq.h
If I want to define this function in my main.c file I will need to include "util/memq.h". This header file is however not available to the application.
```
/home/joakim/repos/zephyrproject/zephyr/samples/bluetooth/peripheral/src/main.c:29:10: fatal error: util/memq.h: No such file or directory
 #include "util/memq.h"
```
So currently, it seems that it is required to be implemented as static inline.

Reply:
Thanks I see what you are saying, In that case you are right. The idea is to implement this in ull_vendor.h. Another usecase is if you are doing proprietary stuff (out of tree) you would have access to these files